packages/frontend/core/src/mobile/pages/workspace/detail/mobile-detail-page.tsx

```diff
@@ -253,9 +253,15 @@ const MobileDetailPage = ({
 
   const allJournalDates = useLiveData(journalService.allJournalDates$);
 
+  const location = useLiveData(workbench.location$);
+  const fromTab = location.search.includes('fromTab');
+
   const handleDateChange = useCallback(
     (date: string) => {
-      openJournal(date);
+      openJournal(date, false, false, {
+        keepSearch: true,
+        replaceHistory: true,
+      });
```

Comment:
just use `fromTab` ?

```suggestion
      openJournal(date, false, fromTab, {
        replaceHistory: true,
      });
```

Reply:
For this case:

- open doc in home
- @ a journal and navigate to that journal
- switch date in date picker

without this, the back button will disappear.